### DISCUSSION TITLE
Reverse Geocoding....one point at a time.

### DISCUSSION THREAD

**Discussion by dietrichmd** [Freika/dawarich#521] [Q&A]
Despite having 4 sidekiq instances available, Dawarich is only doing one reverse geocoding lookup at a time....at least according to Sidekiq.  Is there a way to force dawarich to process more than 1 or at most 2 (I've seen 2 points processed once or twice, but rarely) points at a time?  Or is it actually doing it and sidekiq isn't fast enough to register it?

Looking at htop on the machine with the reverse-geocoding service, I see it only spin up one photon process every minute or so, so I'm confused as to what is going on.

Thanks

**Comment by Freika** [comment]
Nominatim limits are 1 request per second per instance: https://operations.osmfoundation.org/policies/nominatim/. Dawarich is compliant with this limitation.

To bypass it you may want to setup your own instance of reverse geocoding service: https://dawarich.app/docs/tutorials/reverse-geocoding#setting-up-your-own-reverse-geocoding-service

**Comment by dietrichmd** [comment]
Right, I have 3 sidekiq continers set to point to my own self-hosted geocoding service and 1 set to use the default as a fallback..  Yet, only one is active at any given time.  They all work as I've seen them all in use, but never 4 at one time.

Here is my current status page as an example:

![image](https://github.com/user-attachments/assets/2bac57f2-cd08-484e-a99b-53edf4fa434d)

As you can see, only one 'job' is running at any given time.  Also, in this example, the thread being used is a self-hosted one.

**Reply by Freika** [reply]
Might be a problem with rate limiting implementation, I may look into it, but it still would be a lot more beneficial to have your own instance of Photon

**Reply by Freika** [reply]
Although, come to think of it, it works as intended actually: one request per second from a single instance of Dawarich

**Reply by dietrichmd** [reply]
I do have my own instance.  In my screenshot above, the top 3 sidekiq processes use my self-hosted photon, the bottom one is using the default.

**Reply by dietrichmd** [reply]
> Although, come to think of it, it works as intended actually: one request per second from a single instance of Dawarich. 

 I can see this as beneficial for the default service, but those with a self-hosted system should be able to config at least 1 request/sec/instance of sidekiq.

**Reply by Freika** [reply]
Can you check what value is assigned to `PHOTON_API_HOST` inside each of your sidekiq containers?

**Reply by dietrichmd** [reply]
3 of them are using the following:

```
     DISTANCE_UNIT: mi
     #PHOTON_API_HOST: photon.komoot.io
     PHOTON_API_HOST: 192.168.1.5:2322
     PHOTON_API_USE_HTTPS: false
     PROMETHEUS_EXPORTER_ENABLED: false
```

One is using:

```
      DISTANCE_UNIT: mi
      REVERSE_GEOCODING_ENABLED: true
      PHOTON_API_HOST: photon.komoot.io
      #PHOTON_API_HOST: 192.168.1.5:2322
      PHOTON_API_USE_HTTPS: true
```

**Comment by dietrichmd** [comment]
An update....

I changed my docker compose to completely ignore the photogon.komoot.io and only use my local photon....so all 4 sidekiq's are set to local and guess what...now it's processing using all 4.  So it seems if any one of the sidekiq's are set to default, dawarich will only process one geocode/second.  If all are set to local photon, it will use all of them.

![image](https://github.com/user-attachments/assets/18ae73b8-5d2c-4a86-98f5-d1161d46b96d)

**Reply by Freika** [reply]
Yeah, I'll probably have to experiment with this. My guess is that sidekiq containers are reading the value set in the app container, and if this is the case, I'll update docker-compose.yml accordingly and probably note it somewhere.

**Reply by dietrichmd** [reply]
It should be noted that in my case, the app container was set to my local photon and only one sidekiq was set to the default out of the box values -- the rest were set to my local photon.  Once I changed that one sidekiq to local, it just took off!

**Reply by FoxxMD** [reply]
~~@dietrichmd where is this status page located?~~ Found it, it's at `DAWARICH_IP:PORT/sidekiq`
